backend/tests/test_classify_tests.py

```
@@ -0,0 +1,494 @@
+# ---------------------------------------------------------------------------
+# AC-1.4 / AC-1.5 — End-to-end via subprocess: default run excludes tdd_red,
+# unit and integration partitions are disjoint.
+# ---------------------------------------------------------------------------
+
+
+def _run_pytest(args: list[str], cwd: Path) -> subprocess.CompletedProcess:
+    """Invoke pytest as a subprocess against ``cwd``."""
+    env = os.environ.copy()
+    # Avoid pytest discovering OUR test in the child run.
+    env["PYTHONDONTWRITEBYTECODE"] = "1"
+    return subprocess.run(
```

Ruff (0.15.13)

[error] `subprocess` call: check for execution of untrusted input

(S603)